File: csharp/Creational/03-Builder/Builder Interface/IJobCardBuilder.cs
public interface IJobCardBuilder
{
    IJobCardBuilder WithPriority(string priority);
    IJobCardBuilder ScheduledFor(DateTime dateTime);
    IJobCardBuilder WithParts(params string[] parts);
    IJobCardBuilder WithPhotos(params string[] urls);
    IJobCardBuilder WithInstructions(string instructions);
    IJobCardBuilder RequiresSafetyInspection();
    IJobCardBuilder WithWarranty();
    JobCard         Build();
}

File: typescript/Creational/03-Builder/Builder Interface/IJobCardBuilder.ts
import type { JobCard } from "../Product/JobCard";

export interface IJobCardBuilder {
  withPriority(priority: string):        IJobCardBuilder;
  scheduledFor(date: Date):              IJobCardBuilder;
  withParts(...parts: string[]):         IJobCardBuilder;
  withPhotos(...urls: string[]):         IJobCardBuilder;
  withInstructions(text: string):        IJobCardBuilder;
  requiresSafetyInspection():            IJobCardBuilder;
  withWarranty():                        IJobCardBuilder;
  build():                               JobCard;
}